### PR TITLE
Configurable 'Consumer Group' for MessageAnalyzer

### PR DESCRIPTION
**Pull request by vipeller**
Allows injecting 'Consumer Group' name for MessageAnalyzer to use when processing messages.

Usage is: runE2ETest.sh -eventHubConsumerGroup groupname

If not specified, it will inject $Default, which results in the current working, as MessageAnalyzer uses $Default consumer today.

Before specifying any consumer group name, that group name must be created manually in IotHub to make it existing.

### CODE REVIEW

**Comment by philipktlin** [comment]
you need to update LH and stress deployment json files for Windows as well.

File: scripts/linux/runE2ETest.sh [hunk 1]
```diff
@@ -940,6 +946,7 @@ function usage() {
     echo ' -containerRegistryPassword      Password of given username for container registory.'
     echo ' -iotHubConnectionString         IoT hub connection string for creating edge device.'
     echo ' -eventHubConnectionString       Event hub connection string for receive D2C messages.'
+	echo ' -eventHubConsumerGroup          An existing consumer group id for D2C messages.'
```
Comment (philipktlin) [inline scripts/linux/runE2ETest.sh]: nit: spacing.

File: scripts/linux/runE2ETest.sh [hunk 2]
```diff
@@ -347,6 +347,9 @@ function process_args() {
             saveNextArg=0
         elif [ $saveNextArg -eq 28 ]; then
             DPS_MASTER_SYMMETRIC_KEY="$arg"
+            saveNextArg=0
+		elif [ $saveNextArg -eq 29 ]; then
```
Comment (philipktlin) [inline scripts/linux/runE2ETest.sh]: nit: spacing; you can run shellcheck to check style of sh script.

File: scripts/linux/runE2ETest.sh [hunk 3]
```diff
@@ -617,6 +621,7 @@ function run_longhaul_test() {
     local device_id="$RELEASE_LABEL-Linux-$image_architecture_label-longhaul"
 
     sed -i -e "s@<Analyzer.DeviceID>@$device_id@g" "$deployment_working_file"
+	sed -i -e "s@<Analyzer.ConsumerGroupId>@${EVENT_HUB_CONSUMER_GROUP:-\$Default}@g" "$deployment_working_file"
```
Comment (philipktlin) [inline scripts/linux/runE2ETest.sh]: please move this change to prepare_test_from_artifacts function.

File: scripts/linux/runE2ETest.sh [hunk 4]
```diff
@@ -170,6 +170,7 @@ function prepare_test_from_artifacts() {
                 local escapedSnitchAlertUrl
                 local escapedBuildId
                 sed -i -e "s@<Analyzer.EventHubConnectionString>@$EVENTHUB_CONNECTION_STRING@g" "$deployment_working_file"
+                sed -i -e "s@<Analyzer.ConsumerGroupId>@${EVENT_HUB_CONSUMER_GROUP:-\$Default}@g" "$deployment_working_file"
```
Comment (philipktlin) [inline scripts/linux/runE2ETest.sh]: Suggest to move default value assignment below process_args call.

**Comment by philipktlin** [comment]
Please update runE2ETest.ps1 as well; Also, are you going to update stress.yaml and longhaul.yaml to pass different consumer group in this PR or in a separate PR?

**Comment by vipeller** [comment]
I think that should be a separate step (configuring *.yaml). The current change would not cause any differences in behavior, just gives the infrastructure. Then we need to create the appropriate consumer groups in IoTHub and then we can configure it to be used.

**Comment by vipeller** [comment]
merging this now then will do the same with the powershell version in a separate PR to avoid getting obsolate

**Pull request merged**